src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+  const dateArray = date.split(oldSeparator);
+  const dateObject = {};
+
+  for (const i in fromFormat) {
+    let key = fromFormat[i];
+    let value = dateArray[i];
+
+    if (key === 'YYYY' && toFormat.includes('YY')) {
+      key = 'YY';
+      value = [...value].slice(-2).join('');
+    }
+
+    if (key === 'YY' && toFormat.includes('YYYY')) {
+      key = 'YYYY';
+      value = parseInt(value) < 30 ? 20 + value : 19 + value;
```

Comment:
The logic for converting two-digit years to four-digit years is incorrect. The expression `20 + value` and `19 + value` will concatenate strings instead of adding numbers. Use `parseInt(value, 10)` to ensure numeric addition.